doc/man3/EVP_EncryptInit.pod

```diff
@@ -993,6 +993,28 @@ Byte 11-12: Input length (Always 0)
 
 "tls1multi_interleave" must also be set for this operation.
 
+=item "xts_standard" (B<OSSL_CIPHER_PARAM_XTS_STANDARD>) <integer>
+
+Sets the XTS standard to use with SM4-XTS algorithm. XTS mode has two
+implementations, one is standardized in IEEE Std. 1619-2007 and has
+been widely used (e.g., XTS AES), the other is proposed recently
+(GB/T 17964-2021 implemented in May 2022) and is currently only used
+in SM4.
+
+The main difference between them is the multiplication by the
+primitive element E<alpha> to calculate the tweak values. The IEEE
+Std 1619-2007 noted that the multiplication "is a left shift of each
+byte by one bit with carry propagating from one byte to the next
+one", which means that in each byte, the leftmost bit is the most
+significant bit. But in GB/T 17964-2021, the rightmost bit is the
+most significant bit, thus the multiplication becomes a right shift
+of each byte by one bit with carry propagating from one byte to the
+next one. 
+
+The default value is 0, XTS mode of the SM4 algorithm is specified by
+GB/T 17964-2021. Set the parameter to 1 to use the IEEE Std.
+1619-2007 variant of XTS-SM4.
```

Comment:
Considering that the IEEE standard is the default everywhere else, it seems strange to have a different default specifically with SM4.
Yeah, I understand that the GB standard makes sense *regionally*, this gave me pause.

I'm not going to make this a hard rebutal, but I want us to have an extra thought about this.

Reply:
This seemingly odd default is only for SM4. IMO it still makes sense to keep the GB standard the default for SM4 because that is how you're supposed to use SM4 in XTS mode.

If we ever see the need to use the parameter with other ciphers, it would make some sense to switch the values - i.e., 0 for IEEE variant, 1 for GB variant. But we would keep 1 as the default for SM4-XTS.

Would it make it better for you @levitte ?

The other option IMO would be to completely remove the parameter and just use the GB mode with SM4.

Reply:
I like your suggestion where SM4 uses a different default.

Reply:
Thank you very much for raising this issue. Actually, I've also thought a lot about this, 

- Firstly, the parameter should be supported, rather than retaining only the GB mode. GmSSL have implemented the SM4 (alias SMS4) XTS mode specified by IEEE. To avoid data decryption failure after users switch from GmSSL to OpenSSL (if any), the parameter is necessary.
- I think the GB variant should be default, although the standard GB/T 17964-2021 is not mandatory, some certifications require this.